saleor/graphql/csv/mutations.py

```diff
@@ -0,0 +1,141 @@
+from typing import Dict, List, Mapping, Union
+
+import graphene
+from django.core.exceptions import ValidationError
+
+from ...core.permissions import ProductPermissions
+from ...csv import models as csv_models
+from ...csv.events import export_started_event
+from ...csv.utils.export import export_products
+from ..core.enums import CsvErrorCode
+from ..core.mutations import BaseMutation
+from ..core.types.common import CsvError
+from ..product.filters import ProductFilterInput
+from ..product.types import Attribute, Product
+from ..utils import resolve_global_ids_to_primary_keys
+from ..warehouse.types import Warehouse
+from .enums import ExportScope, ProductFieldEnum
+from .types import ExportFile
+
+
+class ExportInfoInput(graphene.InputObjectType):
+    attributes = graphene.List(
+        graphene.NonNull(graphene.ID),
+        description="List of attribute ids witch should be exported.",
+    )
+    warehouses = graphene.List(
+        graphene.NonNull(graphene.ID),
+        description="List of warehouse ids witch should be exported.",
+    )
+    fields = graphene.List(
+        graphene.NonNull(ProductFieldEnum),
+        description="List of product fields witch should be exported.",
+    )
+
+
+class ExportProductsInput(graphene.InputObjectType):
+    scope = ExportScope(
+        description="Determine which products should be exported.", required=True
+    )
+    filter = ProductFilterInput(
+        description="Filtering options for products.", required=False
+    )
+    ids = graphene.List(
+        graphene.NonNull(graphene.ID),
+        description="List of products IDS to export.",
+        required=False,
+    )
+    export_info = ExportInfoInput(
+        description="Input with info about fields which should be exported.",
+        required=True,
+    )
+
+
+class ExportProducts(BaseMutation):
+    export_file = graphene.Field(
+        ExportFile,
+        description=(
+            "The newly created export file job which is responsible for export data."
+        ),
+    )
+
+    class Arguments:
+        input = ExportProductsInput(
+            required=True, description="Fields required to export product data"
+        )
+
+    class Meta:
+        description = "Export products to csv file."
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = CsvError
+        error_type_field = "csv_errors"
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        user = info.context.user
+        input = data["input"]
+        scope = cls.get_products_scope(input)
+        export_info = cls.get_export_info(input["export_info"])
+        export_file = csv_models.ExportFile.objects.create(created_by=user)
+        export_started_event(export_file=export_file, user=user)
```

Comment:
Are you aware that if someone will try to run this mutation with App token, this will raise an unhandled exception?